### PR TITLE
Improve portability of easy_pwait in Readme.me

### PR DESCRIPTION
**Pull request by nickrussler**
Use kill -0 to check if the process is running for POSIX and GNU core utilities compliance

### CODE REVIEW

**Comment by nickrussler** [comment]
Alternatively `ps` could be used, since it would allow checking for processes [not owned by the running user](https://stackoverflow.com/questions/3043978/how-to-check-if-a-process-id-pid-exists#comment12730783_3044045), but ps is not part of the GNU core utilities and [does not seem to be as portable](https://stackoverflow.com/questions/3043978/how-to-check-if-a-process-id-pid-exists#comment64605191_24387818).

**Comment by diazona** [comment]
Thanks for the suggestion! I would kind of like to preserve the ability to wait on a process owned by another user though. I wonder if `ps` with the `-p` or `p` option or some such thing is portable enough? It won't work for every UNIX variant out there, but it probably covers a lot of them.

I wonder if the best approach is to give a couple different versions of `easy_pwait`, e.g. first show one that uses `ps` but then give a `kill`-based backup option, while mentioning the caveat that you can only use it for processes you own.

**Comment by nickrussler** [comment]
I added a `ps -p` based variant

File: README.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
-STOP: you probably want to use something like this
+STOP: Instead you probably want to use something like this
 
     easy_pwait() {
-        while [ -d /proc/$1 ]; do sleep 5; done
+	    while ps -p $1 >/dev/null; do sleep 5; done
```
Comment (diazona) [inline README.md]: whitespace:
```suggestion
            while ps -p $1 >/dev/null; do sleep 5; done
```

File: README.md [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
     }
+	
+or for more portability, which can only check processses owned by the running user
```
Comment (diazona) [inline README.md]:
```suggestion
or a more portable version which can only check processes owned by the running user
```

**Review by diazona — COMMENTED** [review]
The file change looks mostly good, just a couple comments.

Could you also fix the commit messages? One has a subject line that's too long, and in the other there's a typo (`Readme.me` -> `README.md`). Personally, I would suggest squashing the commits together before merging.

**Comment by nickrussler** [comment]
Squashed and accepted your suggestions, although they result in more whitespace than in the original script.

**Comment by diazona** [comment]
Yeah, that's okay. Reducing whitespace is not something I hold as a virtue in general. (Only when it helps readability)

File: README.md [hunk 3]
```diff
@@ -1,10 +1,16 @@
-STOP: you probably want to use something like this
+STOP: Instead you probably want to use something like this
 
     easy_pwait() {
-        while [ -d /proc/$1 ]; do sleep 5; done
+            while ps -p $1 >/dev/null; do sleep 5; done
```
Comment (diazona) [inline README.md]:
```suggestion
        while ps -p $1 >/dev/null; do sleep 5; done
```
Keeping the whitespace consistent between the two functions

File: README.md [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
     }
+
```
Comment (diazona) [inline README.md]:
```suggestion

```
getting rid of trailing whitespace

**Review by diazona — APPROVED** [review]
Just a couple whitespace issues and this is good to go.

**Comment by diazona** [comment]
(sorry for all the noise, I'm not yet used to the new reviewing interface)

**Comment by diazona** [comment]
OK, I merged the changes manually, although I messed it up and Github doesn't recognize it as being merged - sorry! I guess all I can do now is close this. But thanks for the contribution!

**Pull request closed without merging**